### ISSUE TITLE
 improvement(model) : avoid Optional for collection types

### ISSUE THREAD

**Issue by lburgazzoli** [syndesisio/syndesis-rest#520]
Some of the model objects wrap collections with optional , like: 

```java
class Integration {
    Optional<List<? extends Step>> getSteps()
}
```

This patter makes the code a little ugly, like:

```java
integration.getSteps().ifPresent( 
    list -> {
        if (list.isEmpty()) {
            return;
        }

        ...
});
```

I think that collections should not be wrapped by optional.

**Comment by jimmidyson** [comment]
Makes sense to me. We need to make sure that the UI treats empty and null arrays in responses equivalently though - @gashcrumb?

**Comment by lburgazzoli** [comment]
I may have some spare time to work on this, any feedback from UI side @gashcrumb ?

**Comment by zregvart** [comment]
I think this change would be transparent to the UI, i.e. it would not notice the difference, perhaps we should do this for TP2.

**Comment by zregvart** [comment]
@akieling @kahboom @gashcrumb heads up this has been merged. I don't think this should affect the UI but let us know if you see something fishy.

**Comment by KurtStam** [comment]
@zregvart can this be closed now then?

**Comment by zregvart** [comment]
Resolved with #695
